File: adt2sys/sdl/pas/vid_FadeOut.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure vid_FadeOut;

var
  idx: Byte;

function min0(val: Longint): Longint;
begin
  If (val <= 0) then min0 := 0
  else min0 := val;
end;

begin
  For idx := 1 to 15 do
    begin
      rgb_color_alt[idx].r := min0(rgb_color_alt[idx].r-1);
      rgb_color_alt[idx].g := min0(rgb_color_alt[idx].g-1);
      rgb_color_alt[idx].b := min0(rgb_color_alt[idx].b-1);
    end;
  SDL_SetPalette(screen,SDL_PHYSPAL,SDL_ColorArray(Addr(rgb_color_alt)^),0,16);
end;

File: adt2sys/sdl/vid_FadeOut.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void vid_FadeOut (void)
{
  for (uint8_t i = 1; i < 16; i++)
  {
      rgb_color_alt[i].r = rgb_color_alt[i].r - 1 < 0 ? 0 : rgb_color_alt[i].r - 1;
      rgb_color_alt[i].g = rgb_color_alt[i].g - 1 < 0 ? 0 : rgb_color_alt[i].g - 1;
      rgb_color_alt[i].b = rgb_color_alt[i].b - 1 < 0 ? 0 : rgb_color_alt[i].b - 1;
  }
  SDL_SetPalette (screen, SDL_PHYSPAL, (SDL_Color *) &rgb_color_alt, 0, 16);
}
